### PR TITLE
Refactor deployment proposal

### PR DESCRIPTION
**Pull request by fedgiac**
In the code we use the generic term "proposal" to refer to the deployment proposal. As we want to introduce another proposal to make the vCOW token swappable, we need to make the terminology more specific.

This PR does not change the behavior of the scripts.
However, it does change the positions of some types and it fixes a small issue with a test that was skipped in coverage.

### Test Plan

Existing unit tests.

### CODE REVIEW

File: test/deployment-proposal.test.ts [hunk 1]
```diff
@@ -659,8 +659,7 @@ describe("proposal", function () {
         gasPrice: 545019933,
         gasLimit,
       };
-      const signed = await ambExecutor.signTransaction(tx);
-      await hre.ethers.provider.sendTransaction(signed);
+      await ambExecutor.sendTransaction(tx);
```
Comment (fedgiac) [inline test/deployment-proposal.test.ts:662]: I noticed that this test failed when adding new tests in completely unrelated files.
Probably, signing a transaction and then executing it somehow breaks the nonce management in Hardhat tests.

Comment (josojo) [reply test/deployment-proposal.test.ts:662]: Interesting. Good find.

**Review by josojo — APPROVED** [review]
thanks for providing this PR as a separate PR. Makes the review so much easier.
